osconfig_tests/test_suites/package_management/package_management.go

```diff
@@ -112,44 +108,197 @@ func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junit
 	logger.Printf("Finished TestSuite %q", testSuite.Name)
 }
 
-func runCreateOsConfigTest(ctx context.Context, testCase *junitxml.TestCase, logger *log.Logger) {
+func runCreateOsConfigTest(ctx context.Context, testCase *junitxml.TestCase, testSetup *packageManagementTestSetup, logger *log.Logger) {
 
-	osConfig := &osconfigpb.OsConfig{
-		Name: "create-osconfig-test",
+	oc := &osconfigserver.OsConfig{
+		&osconfigpb.OsConfig{
+			Name: "createosconfig-test-osconfig",
+		},
 	}
 
-	defer osconfigserver.CleanupOsConfig(ctx, testCase, logger, osConfig)
+	logger.Printf("create osconfig request:\n%s\n\n", dump.Sprint(oc))
 
-	req := &osconfigpb.CreateOsConfigRequest{
-		Parent:   "projects/compute-image-test-pool-001",
-		OsConfig: osConfig,
-	}
+	parent := fmt.Sprintf("projects/%s", testProject)
+	res, err := osconfigserver.CreateOsConfig(ctx, logger, oc, parent)
 
-	logger.Printf("create osconfig request:\n%s\n\n", dump.Sprint(req))
+	defer cleanuposconfig(ctx, testCase, logger, oc)
 
-	res, err := osconfigserver.CreateOsConfig(ctx, logger, req)
 	if err != nil {
 		testCase.WriteFailure("error while creating osconfig:\n%s\n\n", err)
 	}
 
 	logger.Printf("CreateOsConfig response:\n%s\n\n", dump.Sprint(res))
 }
 
+func runPackageInstallTest(ctx context.Context, testCase *junitxml.TestCase, testSetup *packageManagementTestSetup, logger *log.Logger) {
+
+	osConfig, err := osconfigserver.JsonToOsConfig(packageInstallTestOsConfigString, logger)
+
+	if err != nil {
+		testCase.WriteFailure("error while converting json to osconfig: \n%s\n", err)
+		return
+	}
+
+	oc := &osconfigserver.OsConfig{OsConfig: osConfig}
+
+	parent := fmt.Sprintf("projects/%s", testProject)
+	osconfigserver.CreateOsConfig(ctx, logger, oc, parent)
+
+	if err != nil {
+		testCase.WriteFailure("error while creating osconfig: \n%s\n", err)
+		return
+	}
+
+	defer cleanuposconfig(ctx, testCase, logger, oc)
+
+	assignment, err := osconfigserver.JsonToAssignment(packageInstallTestAssignmentString, logger)
+	if err != nil {
+		testCase.WriteFailure("error while converting json to assignment: \n%s\n", err)
+		return
+	}
+
+	assign := &osconfigserver.Assignment{Assignment: assignment}
+
+	res, err := osconfigserver.CreateAssignment(ctx, logger, assign, parent)
+	if err != nil {
+		testCase.WriteFailure("error while creating assignment: \n%s\n", err)
+	}
+	_ = res
+
+	defer cleanupassignment(ctx, testCase, logger, assign)
+
+	client, err := daisyCompute.NewClient(ctx)
+	if err != nil {
+		testCase.WriteFailure("error creating client: %v", err)
+		return
+	}
+
+	testCase.Logf("Creating instance with image %q", testSetup.image)
+	testSetup.name = fmt.Sprintf("osconfig-test-%s-%s", path.Base(testSetup.image), "packageinstalltest")
```

Comment:
Does `packageinstalltest` have to be one lowercase word? It'd be easier to read if it was camel case or used hyphens. Also, why use a constant string as the variable for `fmt.Sprintf()`? Maybe this can be rewritten as

```
testSetup.name = fmt.Sprintf("osconfig-test-%s-package-install", path.Base(testSetup.image))
```

Reply:
plan was to move this out to a common method once  i add more test cases, since it will have common prefix. for now i am changing it to as you suggested. will move it out in a different commit.